GraphController.cpp

```diff
@@ -29,19 +29,39 @@ void GraphController::handleGet(http_request message)
     {
         if (path[0] == to_string_t("graph")) 
         {
+            if (path.size() < 2)
+            {
+                message.reply(status_codes::BadRequest);
+                return;
+            }
             int id = stoi(path[1]);
             auto graph = graphService->getGraph(id);
+            if (graph == nullptr)
+            {
+                message.reply(status_codes::BadRequest);
```

Comment:
Dla nieistniejącego grafu lepiej by było zwracać NotFound, ale to już kosmetyczna sprawa